src/shell-window-tracker.c

```diff
@@ -739,6 +830,10 @@ shell_window_tracker_init (ShellWindowTracker *self)
 
   self->launched_pid_to_app = g_hash_table_new_full (NULL, NULL, NULL, (GDestroyNotify) g_object_unref);
 
+  self->builder_to_app = g_hash_table_new_full (NULL, NULL, NULL, (GDestroyNotify) g_object_unref);
```

Comment:
Thanks a lot for this find! This fixes the crashers that we have been seeing. 

So we call the untrack_coding_app_window from the shell code for all of the occasions when the window does gets destroyed. That should be handled already.